### ISSUE TITLE
UIComponentBase internal data structure typo

### ISSUE THREAD

**Issue by pizzi80** [eclipse-ee4j/mojarra#5277]
**Describe the bug**

inside `UIComponentBase.java`

there is a typo _"compnent"_ instead of _"component"_

can I fix it? 

or it's a bugged property of Faces API?

```
/*
 * If we can find a valid FacesContext we are going to use it to get access to the property descriptor map.
 */
if (facesContext != null && facesContext.getExternalContext() != null && facesContext.getExternalContext().getApplicationMap() != null) {

    Map<String, Object> applicationMap = facesContext.getExternalContext().getApplicationMap();

    if (!applicationMap.containsKey("com.sun.faces.compnent.COMPONENT_DESCRIPTORS_MAP")) {
        applicationMap.put("com.sun.faces.compnent.COMPONENT_DESCRIPTORS_MAP", new ConcurrentHashMap<>());
    }

    descriptors = (Map<Class<?>, Map<String, PropertyDescriptor>>) 
    applicationMap.get("com.sun.faces.compnent.COMPONENT_DESCRIPTORS_MAP");
    propertyDescriptorMap = descriptors.get(clazz);
}
```

**Comment by BalusC** [comment]
You can fix it. It's not part of public API and not per definition a bug as it isn't used anywhere else with the "correct" key. Can you make it a private constant while at it?

**Comment by pizzi80** [comment]
:+1: :+1:

**Comment by BalusC** [comment]
Already merged into 5.0.
